Integration/NonFreeEnhancerTrait.php

```diff
@@ -4,23 +4,99 @@
 
 trait NonFreeEnhancerTrait
 {
+    /**
+     * @var bool $autorun_enabled Whether or not to run this automatically at Lead creation or only when actively pushed, must be added to the required field array
+     */
+    protected $autorun_enabled = false;
+    
+    /**
+     * @var string|float $cost_per_enhancement Really a decimal type representing the currency cost of the enhancer
+     */
+    protected $cost_per_enhancement = "0.0000";
+    
+    /**
+     * $autorun_enabled getter
+     *
+     * @return bool
+     */
+    public function getAutorunEnabled()
+    {
+        return $this->autorun_enabled;
+    }
+    
+    /**
+     * $autorun_enabled setter
+     *
+     * @param bool $enabled state to set $autorun_enabled to
+     *
+     * @return $this
+     */
+    public function setAutorunEnabled(bool $enabled)
+    {
+       $this->autorun_enabled = $rnabled;
+       
+       return $this;
+    }
+    
+    /**
+     * $cost_per_enhancement getter
+     *
+     * @return string|float
+     */
     public function getCostPerEnhancement()
     {
-        $keys = $this->getDecryptedApiKeys();
-        
-        return $keys['cpe'];
+        return $this->cost_per_enhancement;
```

Comment:
For consistency, we should try to stick to camelCase for variables per Symfony 2 standards.